encoding/kzg/committer/doc.go

```diff
@@ -0,0 +1,78 @@
+//
+// Here's an example scenario which shows that EigenDA could go wrong without this Length.
+// In the extreme case, a malicious disperser could just tell the validators that the blob size is 1,
+// and ask all validators except for one to sign off on the commitment. For a slightly more involved but
+// analogous scenario, assume a network of 8 DA nodes with uniform stake distribution, and coding ratio 1/2.
+// For a blob containing 128 field elements (FEs), each node gets 128*2/8=32 FEs, meaning that any 4 nodes can
+// join forces and reconstruct the data. Now assume a world without length proof; a malicious disperser colluding
+// with a client disperses the same blob/commitment, but claims that the blob only has length of 4 FEs.
+// He sends each node 4*2/8=1 FE. The chunks submitted to the nodes match the commitment, so the nodes accept
+// and sign over the blob’s batch. But now there are only 8 FEs in the system, which is not enough to reconstruct
+// the original blob (need at least 128 for that).
+//
+// ----- Length Commitment + Length Proof Explanation -----
+//
+// In theory, proving an upper bound on the actual blob length is very simple (assuming knowledge of pairings),
+// and would require only a LengthCommitment (no LengthProof needed).
+// - G1 and G2: generators of the bn254 curve groups
+// - BL: blob length (power of 2)
+// - BC_G1: blob commitment; [p(x)]_1 := p(s)G1 (this is the same as our [encoding.BlobCommitments].Commitment)
+// - LC_G1: len commitment; q(x) = x^(2^28-BL)*p(x)
+// Verification is simply e(BC_G1, s^(2^28-BL)*G2) = e(LC_G1, G2)
+//
+// Unfortunately, this simple strategy does not work, due to our (unfortunate) choice of SRS ceremony,
+// which generated 2^29 G1 points but only 2^28 G2 points. Note that this is somehow not documented in
+// https://github.com/privacy-ethereum/perpetualpowersoftau/tree/master itself for some unknown reason...
+// but one can see that there are twice as many points in g1.point than in g2.point from the parsing code, e.g.
+// https://github.com/iden3/snarkjs/blob/e0c7219bd69db078/src/powersoftau_challenge_contribute.js#L22
+// Because of these extra available G1 points, a malicious client/disperser is able to claim that its blob
+// is smaller than it really is, and it can generate a LC_G1 commitment for that smaller blob length,
+// given the extra available G1 SRS points.
+//
+// Attack in practice:
+// - BL: actual blob length, same as above
+// - BC_G1: same as above
+// - FBL: fake blob length = BL/2
+// - FLC_G1: fake length commitment to q'(x) = x^(2^28-BL/2)*p(x)
+// Note that if there were only 2^28 G1 points, then the malicious client/disperser would not be able to generate
+// the commitment FLC_G1, because it has degree 2^28-BL/2+BL = 2^28+BL/2 > 2^28
+// - Verification works: e(BC_G1, s^{2^28-BL/2}G2) = e(FLC_G1, G2)
+//
+// So our actual implementation is as follows:
+// - C1 (G1): blob commitment
```

Comment:
The notation changing is kind of confusing, why not stick to the orignal variables like BC_G1 etc.